### PR TITLE
Bump artifacts action version

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

This PR bypasses dependabot update in an attempt to speed up documentation in the effort for #5352.  v4 of `upload-artifact` and `download-artifact` are reported to be faster, up to 90% improvement in some cases.

### Details

See https://github.com/actions/upload-artifact?tab=readme-ov-file#v4---whats-new

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
A point of reference is that [a recent run](https://github.com/pyvista/pyvista/actions/runs/7247369207) took longer than 31 minutes over two upload steps.

**Comment by MatthewFlamm** [comment]
This PR, the upload takes only 3.5 minutes.  It turns out that our usage is near the 'worst case' reported for `upload-artifact@v3`

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by tkoyama010** [comment]
For no one to be surprised. Since the code owner was set up in #5312, this PR cannot be merged until the code owner approves it.

**Review by banesullivan — APPROVED** [review]
Is that accurate?! 30m down to 2min?!!!

**Comment by MatthewFlamm** [comment]
Accurate!

**Pull request merged**
